### DISCUSSION TITLE
간단한 html css js 프로젝트

### DISCUSSION THREAD

**Discussion by michoball** [wanted-9th-3team/frontend-study#68] [프론트 기술 질문 공간]
저번에 html에 직접 작성하는거에 대해 이야기한게 생각나서 예전에 강의보며 간단한게 만든거 가져와봤습니다. 

![bandicam-2023-06-20-15-09-49-409](https://github.com/wanted-9th-3team/frontend-study/assets/79836148/497f725c-d1d0-40cd-903a-9cc7647eb1fb)

위와 같은 기능을 구현해주시면 됩니다. 
css랑 html zip 파일 첨부하니 안에 있는 html 파일의 script 태그안에 코드 작성해주시면 됩니다. 

잘 작동되면 script 안의 작성하신 코드를 답변으로 달아주세요

문제가 있으면 질문달아주세요 ~

[liveUserFilter.zip](https://github.com/wanted-9th-3team/frontend-study/files/11796781/liveUserFilter.zip)

**Comment by javascriptbiggosoo** [comment]
https://codesandbox.io/s/reverent-aj-2s4ysg?file=/index.html:0-2340

**Comment by michoball** [comment]
https://codesandbox.io/s/wonderful-meninsky-qyj3ms?file=/index.html

**Comment by BaekYuri** [comment]
const filterInput = document.querySelector("#filter");
      const resultUl = document.querySelector("#result");

      let initDataArray = [];
      let resultUlInnerHtml = "";

      let getList = new XMLHttpRequest();

      getList.open("GET", "https://randomuser.me/api?results=50", true);

      getList.onreadystatechange = function (e) {
        const { target } = event;

        if (target.readyState === XMLHttpRequest.DONE) {
          const { status } = target;

          if (status === 0 || (status >= 200 && status < 400)) {
            // 요청이 정상적으로 처리 된 경우
            initDataArray = JSON.parse(target.response).results;

            for (let i = 0; i < initDataArray.length; i++) {
              resultUlInnerHtml += `<li>
              <img src="${initDataArray[i].picture.large}" alt="${initDataArray[i].name.first}">
              <div class="user-info">
                <h4>${initDataArray[i].name.first} ${initDataArray[i].name.last}</h4>
                <p>${initDataArray[i].location.city} , ${initDataArray[i].location.country}</p>
              </div>
              </li>
              `;
            }
            resultUl.innerHTML = resultUlInnerHtml;
            // resultUl.
          } else {
            // 에러가 발생한 경우
            new Error();
          }
        }
      };

      getList.send();

      let filteredData = [];

      filterInput.oninput = (event) => {
        resultUlInnerHtml = "";
        filteredData = initDataArray.filter((user) =>
          `${user.name.first}${user.name.last}`
            .toLowerCase()
            .includes(event.target.value)
        );

        function comparator(a, b) {
          return (
            `${a.name.first} ${a.name.last}`.toLowerCase() <
            `${b.name.first} ${b.name.last}`.toLowerCase()
          );
        }

        filteredData.sort(comparator);
        filteredData.forEach((user) => {
          resultUlInnerHtml += `<li>
              <img src="${user.picture.large}" alt="${user.name.first}">
              <div class="user-info">
                <h4>${user.name.first} ${user.name.last}</h4>
                <p>${user.location.city} , ${user.location.country}</p>
              </div>
              </li>
              `;
        });

        resultUl.innerHTML = resultUlInnerHtml;
      };
